schemas/EGA.common-definitions.json

```diff
@@ -0,0 +1,531 @@
+          }
+        },
+        "anyOf": [
+          { "required": ["alias"]},
+          { "required": ["accession"]}
+        ]
+
+      },
+
+      "custom_attribute": {
+        "type": "object",
+        "title": "Custom attribute of an object",
+        "description": "Reusable attributes to encode tag-value pairs (e.g. Tag being 'Age' and its Value '40') with optional units (e.g. 'years').",
+        "required": ["tag", "value"],
+        "additionalProperties": false,
+        "properties": {
+          "tag": {
+            "type": "string",
+            "title": "Tag of the custom attribute",
+            "description": "The name of the attribute (e.g. 'Age')."
+          },
+          "value": {
+            "type": ["string", "number"],
+            "title": "Value of the custom attribute",
+            "description": "The value of the attribute (e.g. '40')."
+          },
+          "units": {
+            "type": "string",
+            "title": "Units of the custom attribute",
+            "description": "The optional units of the attribute (e.g. 'years')."
+          }
+        }        
+      },
+
+      "file_object": {
+        "type": "object",
+        "title": "EGA File object",
+        "description": "Object containing the base metadata attributes of a file object in the EGA. These can inherited elsewhere with or without extending them.",
+        "required": ["filename", "filetype", "checksum_method", "unencrypted_checksum", "encrypted_checksum"],
+        "additionalProperties": false,
+        "properties": {
+          "filename": {
+            "type": "string",
+            "title": "Filename [data:1050]",
+            "description": "The full name of a file, including all of their file extensions (e.g. .gpg, .md5...), that identifies the file (e.g. 'my-bam-file.bam.gpg').",
+            "pattern": "^[^<>:;,?\"*|/]+$"
+          },
+          "filetype": {
+            "type": "string",
+            "title": "Filetype [NCIT:C172272]",
+            "description": "The nature of the content stored in an electronic file, chosen from a list of CV (e.g. CEL [EFO:0005630] or TSV [NCIT:C164049]).",
+            "enum": ["CEL [EFO:0005630]", "TSV [NCIT:C164049]", "ADF [NCIT:C172213]", "FASTQ [EFO:0004155]", "FASTA [NCIT:C47845]", "SDRF [NCIT:C172211]", "IDF [NCIT:C172212]", "VCF [NCIT:C172216]", "SRA [format:3698]", "SRF [EFO:0004154]", "SFF [EFO:0004156]", "BAM [EFO:0004157]", "CRAM [format:3462]", "XLSX [format:3620]", "CSV [format:3752]", "BED [format:3003]", "IDAT [format:3578 ]", "MAP [format:3285]", "PED [format:3286]", "BIM", "FAM"]
+          },
+          "checksum_method": {
+            "type": "string",
+            "title": "Checksum method [ChecksumAlgorithm]",
```

Comment:
Only the checksum method, or all other nodes in which I joined the strings and ontologized terms? 

For instance, ``filetypes``, instead of:
```` json
"filetype": {
            "type": "string",
            "title": "Filetype [NCIT:C172272]",
            "description": "The nature of the content stored in an electronic file, chosen from a list of CV (e.g. CEL [EFO:0005630] or TSV [NCIT:C164049]).",
            "enum": ["CEL [EFO:0005630]", "TSV [NCIT:C164049]", "ADF [NCIT:C172213]", "FASTQ [EFO:0004155]", "FASTA [NCIT:C47845]", "SDRF [NCIT:C172211]", "IDF [NCIT:C172212]", "VCF [NCIT:C172216]", "SRA [format:3698]", "SRF [EFO:0004154]", "SFF [EFO:0004156]", "BAM [EFO:0004157]", "CRAM [format:3462]", "XLSX [format:3620]", "CSV [format:3752]", "BED [format:3003]", "IDAT [format:3578 ]", "MAP [format:3285]", "PED [format:3286]", "BIM", "FAM"]
          },
````
Something like:
```` json
"filetype": {
            "type": "string",
            "title": "Filetype [NCIT:C172272]",
            "properties": {
              "filetype_name": {
                "enum": ["CEL", "TSV", "ADF", "FASTQ", "FASTA", "SDRF", "IDF", "VCF", "SRA", "SRF", "SFF", "BAM", "CRAM", "XLSX", "CSV", "BED", "IDAT", "MAP", "PED", "BIM", "FAM"]
              },
              "filetype_CURIE": {
                "enum": ["EFO:0005630", "NCIT:C164049", "NCIT:C172213", "EFO:0004155", "NCIT:C47845", "NCIT:C172211", "NCIT:C172212", "NCIT:C172216", "format:3698", "EFO:0004154", "EFO:0004156", "EFO:0004157", "format:3462", "format:3620", "format:3752", "format:3003", "format:3578 ", "format:3285", "format:3286"]
              }
            }            
          }
````